### ISSUE TITLE
Include all Type's Metadata by Default for Types that are Registered for Reflection

### ISSUE THREAD

**Issue by vjovanov** [oracle/graalvm-reachability-metadata#428]
### TL;DR

We'd like to include all the type's metadata for types that are registered for reflection. The newly included metadata would contain method signatures, inner classes, field metadata, record components, signers, and permitted subclasses. The transitively included classes would be included in the image, but not registered for reflection.

This is inspired by [#7753](https://github.com/oracle/graal/issues/7753) where we show that roughly 25% of registration entries can be removed when we use the proposed approach here.

Since this is a breaking change, we propose a new schema for inclusion:
```json
{
    "type": "app.App"
}
```
as this is anyhow required for [proxies and lambdas](https://github.com/oracle/graal/issues/7476). The new entry would be equivalent to
```json
{
    "name": "app.App",
    "allDeclaredClasses": true,
    "allDeclaredMethods": true,
    "allDeclaredFields": true, 
    "allDeclaredConstructors": true,
    "allPublicClasses": true,
    "allPublicMethods": true,
    "allPublicFields": true,
    "allPublicConstructors": true,
    "allRecordComponents": true,
    "allNestMembers": true,
    "allSigners": true,
    "allPermittedSubclasses": true,
    "queryAllDeclaredMethods": true,
    "queryAllDeclaredConstructors": true,
    "queryAllPublicMethods": true,
    "queryAllPublicConstructors": true
}
```
Since this will be changed quickly in the libraries, we intend to backport this change to GraalVM for JDK 17 and JDK 21.

The format with `name` will be kept functional for a long time due to the number of existing projects that use it. However, the agent will be changed to produce the format with `type`.

### Goals
1. Reduce the load for maintaining reachability metadata.
2. Pave the way for inclusion of metadata from user-code. 

### Non-Goals
1. Increase image size significantly--10% or more.
2. Reduce any of the other performance metrics for Native Image.

### Discussion points
1. Should `Class.forName("literal.Type")` include all class metadata in the image?
2. Should we include the default constructors?
3. When merging old and new configs, should the agent (or `native-image-configure`) subsume the previous config with `type`?

**Comment by vjovanov** [comment]
Opened in the wrong repo.
